casanovo/casanovo.py

```diff
@@ -163,8 +163,11 @@ def main(
     }
     # Add extra configuration options and scale by the number of GPUs.
     n_gpus = torch.cuda.device_count()
-    config["n_workers"] = len(psutil.Process().cpu_affinity()) // n_gpus
-    config["train_batch_size"] = config["train_batch_size"] // n_gpus
+    if n_gpus:
+        config["n_workers"] = len(psutil.Process().cpu_affinity()) // n_gpus
+        config["train_batch_size"] = config["train_batch_size"] // n_gpus
+    else:
+        config["n_workers"] = 0
```

Comment:
@bittremieux - I got a div by zero error when running only on CPU. Do you think using 0 workers by default on CPU makes sense? I haven't really looked how number of workers affects inference on CPU. Maybe 1 is better?

Reply:
Rather than 0 or 1, for optimal use of the system resources I'd set to `len(psutil.Process().cpu_affinity())` instead. Dividing by the number of GPUs is only because Pytorch Lightning starts multiple independent instances of the code when running on multiple GPUs.

So I'd change these lines like this:
```
config["n_workers"] = len(psutil.Process().cpu_affinity())
if n_gpus > 1:
    config["n_workers"] = config["n_workers"] // n_gpus
    config["train_batch_size"] = config["train_batch_size"] // n_gpus
```
That should work for both CPU and (multiple) GPUs.